contrib/endpoints/src/api_manager/proto/server_config.proto

```diff
@@ -44,6 +44,12 @@ message ServerConfig {
   // The service name.
   string service_name = 8;
 
+  // Maps service configuration files to their corresponding traffic percentage.
+  // Key is the service configuration file, Value is the traffic percentage
+  // which must be greater than 0.0 and the sum must equal to 100.0.
+  map<string, double> service_configs = 12;
```

Comment:
I prefer to define a message and use repeated field.  call it 

init_service_configs